games/static/games/proto.js

```diff
@@ -1,24 +1,48 @@
 class Game {
-    constructor() {
-
+    constructor(setup) {
+        this.setup = setup
+    }
+    start() {
+        let alienMovementId
+        let grid = document.querySelector('.grid')
+        let board = new Board(grid, 15, this.setup)
+        board.draw(this.setup)
+        this.controls(board)
+        alienMovementId = setInterval(() => { 
+            if (board.alienLocations.length === 0) clearInterval(alienMovementId)
+            board.moveAliens() }, 500)
+        const startButton = document.querySelector('#level1')
+        startButton.addEventListener('mousedown', () => {
+            alienMovementId = setInterval(() => { board.moveAliens() }, 500)
+        })
+        const stopButton = document.querySelector('#level2')
+        stopButton.addEventListener('mousedown', () => {
+            clearInterval(alienMovementId)
+        })
     }
-    controls() {
+    controls(board) {
         document.addEventListener('keydown', (e) => {
-            if(e.keyCode === 37) {
+            if (e.key === 'ArrowLeft') {
                 board.moveDefender('left')
-            } else if (e.keyCode === 39) {
+            }
+            if (e.key === 'ArrowRight') {
                 board.moveDefender('right')
-            } 
-            if (e.keyCode === 32) board.shootRocket(e) 
+            }
+
+        })
+        document.addEventListener('keyup', (e) => {
+            if (e.key === ' ') {
+                board.shootRocket()
+            }
         })
     }
+
 }
 
 class Setup {
-    constructor(alienLocations, gameSpeed, movementPattern) {
+    constructor(alienLocations, gameSpeed) {
```

Comment:
you removed movementPattern but still got the movement pattern method.
and basicly if theres no behaviour to this class it can just be a data object (so you can remove the methods and use the fields directly. like:
`setup.gameSpeed`

Reply:
i actually would like it to receive a movement pattern now that i am using a movement pattern but i wonder if it is a good time to do it, maybe add it as a mission on the board as this branch is mainly for the movement of the rockets

Reply:
sounds good